crates/db_views_actor/src/person_view.rs

```diff
@@ -101,45 +102,76 @@ impl PersonQuery {
-      SortType::TopThreeMonths => query
-        .filter(person::published.gt(now - 3.months()))
-        .order_by(person_aggregates::comment_score.desc()),
-      SortType::TopSixMonths => query
-        .filter(person::published.gt(now - 6.months()))
-        .order_by(person_aggregates::comment_score.desc()),
-      SortType::TopNineMonths => query
-        .filter(person::published.gt(now - 9.months()))
-        .order_by(person_aggregates::comment_score.desc()),
+      SortType::TopWeek | SortType::BestWeek => query.filter(person::published.gt(now - 1.weeks())),
+      SortType::TopDay | SortType::BestDay => query.filter(person::published.gt(now - 1.days())),
+      SortType::TopSixHour | SortType::BestSixHour => {
+        query.filter(person::published.gt(now - 6.hours()))
+      }
+      SortType::TopThreeMonths | SortType::BestThreeMonths => {
+        query.filter(person::published.gt(now - 3.months()))
+      }
+      SortType::TopSixMonths | SortType::BestSixMonths => {
+        query.filter(person::published.gt(now - 6.months()))
+      }
+      SortType::TopNineMonths | SortType::BestNineMonths => {
+        query.filter(person::published.gt(now - 9.months()))
+      }
+      SortType::TopTwelveHour | SortType::BestTwelveHour => {
+        query.filter(person::published.gt(now - 12.hours()))
+      }
+      SortType::TopHour | SortType::BestHour => query.filter(person::published.gt(now - 1.hours())),
+
+      _ => query,
     };
 
+    query = match self.sort.unwrap_or(SortType::Hot) {
+            SortType::New | SortType::NewComments => query.order_by(person::published.desc()),
+            SortType::Old => query.order_by(person::published.asc()),
+            SortType::MostComments => query.order_by(person_aggregates::comment_count.desc()),
+
+          SortType::Hot |
+          SortType::Active |
+          SortType::TopAll |
+          SortType::TopYear |
+          SortType::TopMonth |
+          SortType::TopWeek |
+          SortType::TopDay |
+          SortType::TopHour |
+          SortType::TopSixHour |
+          SortType::TopTwelveHour |
+          SortType::TopThreeMonths |
+          SortType::TopSixMonths |
+          SortType::TopNineMonths => {
+            query.order_by(person_aggregates::comment_score.desc())
+            .then_order_by(person::published.desc())
+          },
+          SortType::BestAll |
+          SortType::BestYear |
+          SortType::BestThreeMonths |
+          SortType::BestSixMonths |
+          SortType::BestNineMonths |
+          SortType::BestMonth |
+          SortType::BestWeek |
+          SortType::BestDay |
+          SortType::BestTwelveHour |
+          SortType::BestSixHour |
+          SortType::BestHour=> {
+            query
+            .then_order_by(
+                sql::<sql_types::BigInt>(
+                "row_number() OVER (PARTITION BY \"person\".\"instance_id\" ORDER BY \"person_aggregates\".\"comment_score\" DESC)"
```

Comment:
I couldn't even run this with production data, postgres said it ran out of memory.